src/django_idom/utils.py

```diff
@@ -0,0 +1,124 @@
+import logging
+import os
+import re
+from fnmatch import fnmatch
+from importlib import import_module
+from typing import Set
+
+from django.template import engines
+from django.utils.encoding import smart_str
+
+from django_idom.config import IDOM_REGISTERED_COMPONENTS
+
+
+COMPONENT_REGEX_PATTERN = r"{% *idom_component ((\"[^\"']*\")|('[^\"']*')).*?%}"
```

Comment:
You should be able to just do `COMPONENT_REGEX_PATTERN = re.compile(...)` so you don't have to re-compile every time its used.